src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java

```diff
@@ -639,6 +666,24 @@ public String initAndValidate(Logger log) throws IllegalArgumentException {
             healthCheck.validate();
         }
 
+        List<AssemblyConfiguration> assemblyConfigurations = getAssemblyConfigurations();
+        if (!assemblyConfigurations.isEmpty()) {
+            Map<String, Integer> seenNames = new HashMap<>();
+            for (AssemblyConfiguration config : assemblyConfigurations) {
+                seenNames.compute(config.getName(), (k, v) -> v == null ? 1 : v + 1);
+            }
```

Comment:
I see you're trying to find assembly names with the same name. Do you think we can use a `Set` here instead? [Set `add()`](https://docs.oracle.com/javase/6/docs/api/java/util/Set.html#add(E)) method returns false if the element already exists in the collection. I haven't tried this myself but this can be simplified like this:
```java
Set<String> assemblyNames = new HashSet<>();
for (AssemblyConfiguration assemblyConfiguration : assemblyConfigurations) {
    boolean wasElementAbsent = assemblyNames.add(assemblyConfiguration.getName());
    if (!wasElementAbsent) {
        log.error("Multiple assemblies use the name %s. Please assign each assembly a unique name.");
        throw new IllegalArgumentException("Assembly names must be unique");
    }
}
```

Reply:
For as long as I've been using Java, I never looked at the return value of Set.add(). This is much nicer.